libssu/ssurepomanager.cpp

```diff
@@ -59,16 +60,20 @@ int SsuRepoManager::add(const QString &repo, const QString &repoUrl)
         return -1;
 
     if (repoUrl.isEmpty()) {
-        // just enable a repository which has URL in repos.ini
+        // Enable a repository if it has URL in repos.ini
         QStringList enabledRepos;
         if (ssuSettings->contains("enabled-repos"))
             enabledRepos = ssuSettings->value("enabled-repos").toStringList();
 
-        enabledRepos.append(repo);
-        enabledRepos.removeDuplicates();
-        ssuSettings->setValue("enabled-repos", enabledRepos);
+        if (globalRepos.contains(repo)) {
+            enabledRepos.append(repo);
+            enabledRepos.removeDuplicates();
+            ssuSettings->setValue("enabled-repos", enabledRepos);
+        }
     } else {
-        ssuSettings->setValue("repository-urls/" + repo, repoUrl);
+        if (!globalRepos.contains(repo)) {
```

Comment:
```suggestion
        else if (!globalRepos.contains(repo)) {
```
You can skip one level of indentation.

Reply:
Done. I wondered whether to report an error in an extra else, but I think that case is dealt with elsewhere.